build.sbt

```diff
@@ -447,6 +447,8 @@ lazy val metals = project
       "com.outr" %% "scribe-slf4j2" % V.scribe, // needed for flyway database migrations
       // for JSON formatted doctor
       "com.lihaoyi" %% "ujson" % "4.2.1",
+      // For MCP communication
+      "com.github.andyglow" %% "scala-jsonschema-circe-json" % "0.7.11",
```

Comment:
Under which conditions is it ok to introduce a new dependency?
The `MetalsMcpServer` became quite large with the set of new tools, so I used this library to generate the JSON Schema for the tools and did a few other refactorings to reduce the amount of boilerplate.

Reply:
We usually try to minimize the amount of dependencies, so if it's not 100% needed we don't add it. I would rather avoid it if possible. We can easily separate the tools into traits and then just have one class to create them all